### PR TITLE
[staging] Correcting multilingual status module when a content language is in the trash

### PR DESCRIPTION
**Pull request by infograf768**
If a content language is trashed but not deleted, it appears as Published in the Multilingual status Module.
Thanks Mig for finding this bug.

### Summary of Changes
Separating the statuses to include the trashed possibility.
Display a trash icon when a Content Language is trashed.
Also display a specific message on top of the module in that case.

### Testing Instructions
Install a multilingual site from a clean Joomla.
Use multiple content languages.
Trash one of the content languages.
Display the module.

fa-IR is trashed below

### Before patch

screen shot 2019-01-30 at 16 38 07

### After patch
screen shot 2019-01-30 at 16 37 02

### CODE REVIEW

**Comment by imanickam** [comment]
I have tested this item :white_check_mark: successfully on 7a1d26ff5bd8de9407e3ea506ff9863fdcd929a6<br /><br />I have tested this successfully on Joomla! 3.9.2.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23722">issues.joomla.org/tracker/joomla-cms/23722</a>.</sub>

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 7a1d26ff5bd8de9407e3ea506ff9863fdcd929a6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23722">issues.joomla.org/tracker/joomla-cms/23722</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23722">issues.joomla.org/tracker/joomla-cms/23722</a>.</sub>

File: administrator/language/en-GB/en-GB.com_languages.ini [hunk 1]
```diff
@@ -92,6 +92,7 @@ COM_LANGUAGES_MULTILANGSTATUS_CONTACTS_ERROR_TIP="Warning! A user/author should
 COM_LANGUAGES_MULTILANGSTATUS_CONTENT_LANGUAGE_PUBLISHED="Published Content Languages"
 COM_LANGUAGES_MULTILANGSTATUS_DEFAULT_HOME_MODULE_PUBLISHED="This site is set as a multilingual site. The menu module displaying the Home menu item set to language &quot;All&quot; should not be published."
 COM_LANGUAGES_MULTILANGSTATUS_ERROR_CONTENT_LANGUAGE="A Default Home page is assigned to the <strong>%s</strong> Content Language although a Site Language for this Content Language is not installed or published AND/OR the Content Language is not published."
+COM_LANGUAGES_MULTILANGSTATUS_ERROR_CONTENT_LANGUAGE_TRASHED="The Content Language <strong>%s</strong> is trashed. Change its status to be able to use it."
```
Comment (HLeithner) [inline administrator/language/en-GB/en-GB.com_languages.ini]: @brianteeman would you check this translation, sounds a bit strange to me.

File: administrator/components/com_languages/views/multilangstatus/tmpl/default.php [hunk 2]
```diff
@@ -84,6 +84,16 @@
 					</td>
 				</tr>
 			<?php endif; ?>
+			<?php if ($contentlang->published == -2) : ?>
```
Comment (HLeithner) [inline administrator/components/com_languages/views/multilangstatus/tmpl/default.php:87]: Can we validate this type safe?

File: administrator/components/com_languages/views/multilangstatus/tmpl/default.php [hunk 3]
```diff
@@ -195,14 +205,18 @@
 						</td>
 				<?php endif; ?>
 				<?php // Published Content languages ?>
-				<?php if ($status->lang_code && $status->published) : ?>
+				<?php if ($status->lang_code && $status->published == 1) : ?>
```
Comment (HLeithner) [inline administrator/components/com_languages/views/multilangstatus/tmpl/default.php:208]: Can we validate this type safe?
I know the code is old but whats the reason to check the $status->lang_code?

File: administrator/components/com_languages/views/multilangstatus/tmpl/default.php [hunk 4]
```diff
@@ -195,14 +205,18 @@
...
 						<td class="center">
 							<span class="icon-ok" aria-hidden="true"></span><span class="element-invisible"><?php echo JText::_('JYES'); ?></span>
 						</td>
-				<?php else : ?>
+				<?php elseif ($status->lang_code && $status->published == 0) : ?>
```
Comment (HLeithner) [inline administrator/components/com_languages/views/multilangstatus/tmpl/default.php:212]: Can we validate this type safe?

File: administrator/components/com_languages/views/multilangstatus/tmpl/default.php [hunk 5]
```diff
@@ -195,14 +205,18 @@
...
 						<td class="center">
 							<span class="icon-pending" aria-hidden="true"></span><span class="element-invisible"><?php echo JText::_('WARNING'); ?></span>
 						</td>
+				<?php elseif ($status->lang_code && $status->published == -2) : ?>
```
Comment (HLeithner) [inline administrator/components/com_languages/views/multilangstatus/tmpl/default.php:216]: Can we validate this type safe?

File: administrator/components/com_languages/views/multilangstatus/tmpl/default.php (hunk 3 above)
Comment (infograf768) [reply administrator/components/com_languages/views/multilangstatus/tmpl/default.php:208]: validated.
I guess we used `$status->lang_code` to be sure something was not wrong in db.

File: administrator/components/com_languages/views/multilangstatus/tmpl/default.php [hunk 6]
```diff
@@ -195,14 +205,18 @@
...
-				<?php else : ?>
+				<?php elseif ($status->lang_code && $status->published === 0) : ?>
 						<td class="center">
 							<span class="icon-pending" aria-hidden="true"></span><span class="element-invisible"><?php echo JText::_('WARNING'); ?></span>
 						</td>
+				<?php elseif ($status->lang_code && $status->published === -2) : ?>
+						<td class="center">
+							<span class="icon-trash" aria-hidden="true"></span><span class="element-invisible"><?php echo JText::_('WARNING'); ?></span>
+						</td>
 				<?php endif; ?>
```
Comment (HLeithner) [inline administrator/components/com_languages/views/multilangstatus/tmpl/default.php]: If we have none of the above states, we have a missing <td></td>

**Comment by HLeithner** [comment]
In Line 216/230 we have 2 closing </td> could you remove one of them.

File: administrator/language/en-GB/en-GB.com_languages.ini (hunk 1 above)
Comment (infograf768) [reply administrator/language/en-GB/en-GB.com_languages.ini]: Here, we must not forget that it is perfectly OK to have an Unpublished Content language as it lets use the language in the language fields to prepare using a new language in the multilingual site until Published to display it in frontend.
Therefore, the only status which create issues is Trashed. It should be deleted or Published/Unpublished.
I am not fully satisfied either with the wording.

File: administrator/components/com_languages/views/multilangstatus/tmpl/default.php (hunk 6 above)
Comment (infograf768) [reply administrator/components/com_languages/views/multilangstatus/tmpl/default.php]: It jumped out of there and took holidays.
back in now...

File: administrator/language/en-GB/en-GB.com_languages.ini (hunk 1 above)
Comment (brianteeman) [reply administrator/language/en-GB/en-GB.com_languages.ini]: if the error goes away if it is unpublished then it is not correct to say "to be able to use it"

Comment (HLeithner) [reply administrator/language/en-GB/en-GB.com_languages.ini]: Do we need the last sentence? Isn't it clear that you have to change the status to fix this problem?

Comment (brianteeman) [reply administrator/language/en-GB/en-GB.com_languages.ini]: I agree.

File: administrator/components/com_languages/views/multilangstatus/tmpl/default.php [hunk 7]
```diff
@@ -195,10 +205,18 @@
...
 						<td class="center">
 							<span class="icon-ok" aria-hidden="true"></span><span class="element-invisible"><?php echo JText::_('JYES'); ?></span>
 						</td>
+				<?php elseif ($status->lang_code && $status->published === 0) : ?>
+						<td class="center">
+							<span class="icon-pending" aria-hidden="true"></span><span class="element-invisible"><?php echo JText::_('WARNING'); ?></span>
+						</td>
+				<?php elseif ($status->lang_code && $status->published === -2) : ?>
+						<td class="center">
+							<span class="icon-trash" aria-hidden="true"></span><span class="element-invisible"><?php echo JText::_('WARNING'); ?></span>
```
Comment (Quy) [inline administrator/components/com_languages/views/multilangstatus/tmpl/default.php:218]: Should the text be `Trashed`?

Comment (Quy) [reply administrator/components/com_languages/views/multilangstatus/tmpl/default.php:218]: Fix markup. Move line 264 `</tr>` after line 261 to be inside the loop.

File: administrator/components/com_languages/views/multilangstatus/tmpl/default.php (hunk 3 above)
Comment (Quy) [reply administrator/components/com_languages/views/multilangstatus/tmpl/default.php:208]: This should be reverted. `$status->published` returns a string value.

**Comment by infograf768** [comment]
Thanks @Quy 
Indeed I should have tested before changing to ===
`</tr>` also corrected (that was an old one)

File: administrator/components/com_languages/views/multilangstatus/tmpl/default.php (hunk 7 above)
Comment (infograf768) [reply administrator/components/com_languages/views/multilangstatus/tmpl/default.php:218]: > Should the text be Trashed

I do not think so, it is a Warning. same for Unpublished.

**Comment by infograf768** [comment]
@imanickam @alikon
Please test again. Same instructions. Policy wants it.

**Comment by HLeithner** [comment]
thx

**Pull request merged**